form-validation.js

```diff
@@ -7,25 +7,29 @@ const messageMobile = document.querySelector('#messageEmail-mobile');
 const nameInput = document.querySelector('.name');
 const emailInput = document.querySelector('.email');
 const messageInput = document.querySelector('.message');
-// const formbutton = document.querySelector('#form-button');
-// const formbuttondesktop = document.querySelector('#form-button-desk');
+const rawData = localStorage.getItem('cachedFormData');
 
 const formobject = {
   name: '',
   email: '',
   message: '',
 };
 
-function storeInput() {
-  const stringobject = JSON.stringify(formobject);
-  nameInput.value = formobject.name;
+if (rawData) {
+  const formobject = JSON.parse(rawData);
   emailInput.value = formobject.email;
+  nameInput.value = formobject.name;
   messageInput.value = formobject.message;
-  localStorage.setItem('formobject', stringobject);
 }
 
-form.addEventListener('click', storeInput());
-mobileForm.addEventListener('click', storeInput());
+const dataStore = (event) => {
+  const element = event.target;
+  formobject[element.name] = element.value;
+  localStorage.setItem('cachedFormData', JSON.stringify(formobject));
+};
+nameInput.addEventListener('change', dataStore);
+emailInput.addEventListener('change', dataStore);
+messageInput.addEventListener('change', dataStore);
```

Comment:
- [ ] Good job so far implementing the local storage, But I think there are some requirements you need to meet before moving to the next milestone. according to the project requirement, User inputs are meant to be saved to the local storage as the user enters it, But currently, this is feature does not work, Please look into this.. :smile: :+1: 
- Screen record video for reference:

https://user-images.githubusercontent.com/107632780/233811137-c665a88a-0c7f-4dc8-92bc-e16f8ada0fd4.mp4